### PR TITLE
feat: inject DynamoDbTable and DynamoDbAsyncTable

### LINKED ISSUE #809: [DynamoDbEnhanced] Support injection of DynamoDbTable<DynamoDBExampleTableEntry>
Being able to inject directly `DynamoDbTable` would reduce boilerplate code to get one.

```java
@Path("/dynamodbenhanced")
public class DynamoDbEnhancedResource {

    private final static String BLOCKING_TABLE = "enhancedblocking";

    @Inject
    @TableName(ASYNC_TABLE)
    DynamoDbTable<DynamoDBExampleTableEntry> exampleBlockingTable;

    @GET
    @Path("blocking")
    @Produces(TEXT_PLAIN)
    public String testBlockingDynamo() {
        LOG.info("Testing Blocking Dynamodb client with table: " + BLOCKING_TABLE);

        String partitionKeyAsString = UUID.randomUUID().toString();
        String rangeId = UUID.randomUUID().toString();

        exampleBlockingTable.createTable();

        DynamoDBExampleTableEntry exampleTableEntry = new DynamoDBExampleTableEntry();
        exampleTableEntry.setKeyId(partitionKeyAsString);
        exampleTableEntry.setRangeId(rangeId);
        exampleTableEntry.setPayload(PAYLOAD_VALUE);

        exampleBlockingTable.putItem(exampleTableEntry);

        Key partitionKey = Key.builder().partitionValue(partitionKeyAsString).build();

        DynamoDBExampleTableEntry existingTableEntry = exampleBlockingTable.getItem(partitionKey);

        if (existingTableEntry != null) {
            return existingTableEntry.getPayload() + "@" + existingTableEntry.getVersion();
        } else {
            return "ERROR";
        }
    }
}
````

### PR DESCRIPTION
**Pull request by scrocquesel**
Closes #809

### CODE REVIEW

**Comment by scrocquesel** [comment]
@knuspertante Would you mind review this PR ?

File: dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/NamedDynamoDbTable.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import jakarta.inject.Qualifier;
+
+/**
+ * Specification of DynamoDb table to be injected.
+ */
+@Target({ FIELD, METHOD, PARAMETER })
```
Comment (knuspertante) [inline dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/NamedDynamoDbTable.java:17]: Is this qualifier applicable to method and parameter? Do we handle this in DynamodbEnhancedDbTableProcessor or anywhere else? If so, sorry, I'm still learning to write quarkus extensions, could be possible that I missed something ;-)

**Review by knuspertante — COMMENTED** [review]
Nice, thank you! LGTM!

File: dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/NamedDynamoDbTable.java (hunk 1 above)
Comment (scrocquesel) [reply dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/NamedDynamoDbTable.java]: @knuspertante Thanks. you are right, injection point are discovered in `DynamodbEnhancedDbTableProcessor` and it only processes fields

**Pull request merged**